docs/FileIO.md

```diff
@@ -0,0 +1,120 @@
+
+## The `FileIO` module {#sec-fileio}
+
+The `FileIO` module provides basic file I/O operations.
+Right now, these are limited to reading bytes from a file and writing bytes to a file.
+The API is intentionally limited in scope and will be expanded later.
+
+Unlike other modules in the `libraries` repository,
+the `FileIO` module will not compile or run correctly without a language-specific implementation file.
+Language-specific implementation files are provided for C#/.NET, Java, and Javascript.
+(Python and Golang support are planned.)
+Concretely, to use `FileIO` in your code, you must:
+
+1. `include` and `import` the `FileIO` module as you would any other library module
+2. incorporate the corresponding language-specific implementation file (e.g. `FileIO.cs`) when building or running your program
```

Comment:
Should we touch on the problem of composing Dafny libraries?
e.g. if you try and use 2 Dafny libraries that both use `FileIO`
you are going to have a bad time.

Reply:
Yes. We have a fair bit of work to do to make a collection of useful code into one or more libraries. Tools need to be able to incorporate into the users compilation just as much from the library as is needed, no more, and no duplication. Something like a linker for Dafny.